src/Frontend/Fragment/Category.hs

```diff
@@ -25,16 +25,16 @@ import qualified Data.Text as ST
 import qualified Lucid
 import qualified Text.Digestive.Form as DF
 import qualified Text.Digestive.Lucid.Html5 as DF
-import qualified Text.Digestive.Types as DF
 
 
 -- | FIXME: 'makeFormSelectCategory', 'formPageSelectCategory' should be a subform.  (related: `grep
 -- subform src/Frontend/Page/Topic.hs`.)
+-- FIXME: Error and non selected category are inseparable cases.
```

Comment:
This could be achieved with a custom `instance Enum a => Enum (Maybe a)` instance:

````haskell
instance Enum a => Maybe (Enum a) where
    toEnum 0 = Nothing
    toEnum n = Just (toEnum (n-1))

    fromEnum Nothing = 0
    fromEnum (Just n) = (fromEnum n) - 1
````

and also some work with the JS code.